### PR TITLE
Adding molecule tests

### PR DESCRIPTION
**Pull request by neilhaggerty**
Also moving dependencies into the task as this is the "new" syntax.

Keeping the geerlingguy.pip dependency in meta/ as we want to pin the
version.

PLAT-1595

### CODE REVIEW

File: molecule/default/molecule.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  name: galaxy
+  role-file: requirements.yml
+driver:
+  name: docker
+lint:
+  name: yamllint
+platforms:
+  - name: centos_image
+    image: centos:7
+    privileged: true
```
Comment (tomyan) [inline molecule/default/molecule.yml]: Is privileged necessary for these two contianers?

File: meta/main.yml [hunk 2]
```diff
@@ -27,27 +28,26 @@ galaxy_info:
...
-    # List tags for your role here, one per line. A tag is a keyword that describes
-    # and categorizes the role. Users find roles by searching for tags. Be sure to
-    # remove the '[]' above, if you add tags to this list.
-    #
-    # NOTE: A tag is limited to a single word comprised of alphanumeric characters.
-    #       Maximum 20 tags per role.
-
-dependencies:
-    - mergermarket.pip
-    - mergermarket.docker
```
Comment (grahamlyons) [inline meta/main.yml:53]: We run the agent in Docker so we need Docker to be installed so we need our Ansible role for Docker, no?

File: molecule/default/tests/test_default.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    assert f.exists
+
+
+def test_datadog_jmx_config_file_exists(host):
+    f = host.file('/etc/datadog-agent/conf.d/jmx.d/conf.yaml')
+    assert f.exists
+
+
+def test_datadog_docker_image_exists(host):
+    output = host.check_output('docker images datadog/agent -q | wc -l')
```
Comment (tomyan) [inline molecule/default/tests/test_default.py:31]: This check seems a little inexact - would match 1 or 947372723282184834, but not 947372723282784834.

File: molecule/default/requirements.yml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+---
+# from galaxy
+- src: geerlingguy.repo-epel
```
Comment (grahamlyons) [inline molecule/default/requirements.yml:3]: We've not got this in the dependencies for the role. Do you know what we pull from EPEL?

**Review by grahamlyons — CHANGES_REQUESTED** [review]
Looks great. Just got a couple of questions.

File: test.sh [hunk 5]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+docker run --rm -it \
+    -v $(pwd):/${PWD##*/}:ro \
```
Comment (tomyan) [inline test.sh]: I find `${PWD##*/}` a little cryptic. Maybe split out into a temporary variable with a descriptive name?

File: test.sh [hunk 6]
```diff
@@ -0,0 +1,8 @@
...
+    -v /var/run/docker.sock:/var/run/docker.sock \
+    -w /${PWD##*/} \
```
Comment (tomyan) [inline test.sh]: ...would also DRY this out a little.

**Review by tomyan — APPROVED** [review]
Minor comments inline

File: meta/main.yml (hunk 2 above)
Comment (grahamlyons) [reply meta/main.yml:53]: Just re-read your description on the pull request about the "new" syntax for dependencies. Does that install them just-in-time then?

**Review by grahamlyons — APPROVED** [review]

File: molecule/default/tests/test_default.py [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    assert '1' in output
```
Comment (grahamlyons) [inline molecule/default/tests/test_default.py]: Perhaps `output.strip() == '1'`?

Comment (grahamlyons) [reply molecule/default/tests/test_default.py]: Or:
```python
output = host.check_output('docker images datadog/agent -q')
lines = output.split('\n')
assert len(lines) == 1
```

Comment (tomyan) [reply molecule/default/tests/test_default.py]: Or:

    lines = int(host.check_output('docker images datadog/agent -q | wc -l'))
    assert lines == 1

File: meta/main.yml (hunk 2 above)
Comment (neilhaggerty) [reply meta/main.yml:53]: Going to move these back and let users of the role decide on how to manage dependencies

File: molecule/default/molecule.yml (hunk 1 above)
Comment (neilhaggerty) [reply molecule/default/molecule.yml]: Have removed

File: molecule/default/tests/test_default.py (hunk 3 above)
Comment (neilhaggerty) [reply molecule/default/tests/test_default.py:31]: I don't understand this

File: test.sh (hunk 6 above)
Comment (neilhaggerty) [reply test.sh]: don't understand this

File: molecule/default/tests/test_default.py (hunk 3 above)
Comment (tomyan) [reply molecule/default/tests/test_default.py:31]: Was just saying (in a bit of a round about way) that it should check it's one not that it contains the digit 1 somewhere.

**Review by tomyan — APPROVED** [review]

File: molecule/default/tests/test_default.py (hunk 3 above)
Comment (neilhaggerty) [reply molecule/default/tests/test_default.py:31]: ohhhh... I didn't realise I was using `in`.. I meant `==` good catch. Have fixed now

... was a little cryptic though haha

**Pull request merged**